src/sentry/eventstore/snuba/backend.py

```diff
@@ -108,9 +110,62 @@ def __get_events(
         should_bind_nodes=False,
     ):
         assert filter, "You must provide a filter"
-        cols = self.__get_columns(additional_columns)
+        cols = self.__get_columns()
         orderby = orderby or DESC_ORDERING
 
+        # This is an optimization for the Group.filter_by_event_id query where we
+        # have a single event ID and want to check all accessible projects for a
+        # direct hit. In this case it's usually faster to go to nodestore first.
+        if (
+            filter.event_ids
+            and filter.project_ids
+            and len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
+            and offset == 0
+        ):
+            event_list = [
+                Event(project_id=project_id, event_id=event_id)
+                for event_id in filter.event_ids
+                for project_id in filter.project_ids
+            ]
+            self.bind_nodes(event_list)
+
+            events = [event for event in event_list if len(event.data)]
+
+            if events:
+                event_ids = {event.event_id for event in events}
+                project_ids = {event.project_id for event in events}
+                start = min(event.datetime for event in events)
+                end = max(event.datetime for event in events) + timedelta(seconds=1)
+
+                result = snuba.aliased_query(
+                    selected_columns=cols,
+                    start=start,
+                    end=end,
+                    conditions=filter.conditions,
+                    filter_keys={"project_id": project_ids, "event_id": event_ids},
+                    orderby=orderby,
+                    limit=len(events),
+                    offset=DEFAULT_OFFSET,
+                    referrer=referrer,
+                    dataset=snuba.Dataset.Events,
+                )
+
+                if "error" not in result:
+                    events = [self.__make_event(evt) for evt in result["data"]]
+
+                    # Bind previously fetched node data
+                    for event in events:
+                        node_data = next(
+                            e
+                            for e in events
```

Comment:
Assuming here you wanted the old events list from nodestore, this is not it. You reassigned it at line 154.

Reply:
Mmmm, I have no idea why I called everything "events"